internal/k8s/minikube_fake.go

```diff
@@ -0,0 +1,17 @@
+package k8s
+
+import "context"
+
+type FakeMinkube struct {
```

Comment:
it seems weird to move this to the k8s package? what's the reasoning behind this change?

Reply:
I ran into some dependency cycle issues because now the k8sClient needs a minikubeClient, but the minikubeClient needs (or at least wants) a `k8s.KubeContext` (ie the cluster name for `-p <clusterName>`).

This also felt kinda weird to me, would be happy for alternate suggestions.

Reply:
ah, i see. OK. I wonder if we're going to have to move the k8s data types to their own package (like kubernetes itself does with v1.Pod, etc). I feel like there are a couple places where we run into this problem.